src/pages/index.js

```diff
@@ -3,49 +3,179 @@ import Card from "../components/Card.js";
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "bdefa1ce-28f3-4e1a-8e75-6e2193fb1dff",
+    "Content-Type": "application/json",
+  },
+});
+
+console.log("API initialized");
+
 // Initialize form validators
 const formValidators = {};
 const enableValidation = (config) => {
   const formList = Array.from(document.querySelectorAll(config.formSelector));
+  console.log("Forms found:", formList.length);
   formList.forEach((formElement) => {
     const validator = new FormValidator(config, formElement);
     const formName = formElement.getAttribute("name");
+    console.log("Registering validator for form:", formName);
     formValidators[formName] = validator;
     validator.enableValidation();
   });
 };
 
 enableValidation(validationSettings);
+console.log("Form validation enabled");
 
 // Initialize UserInfo
 const userInfo = new UserInfo({
   nameSelector: selectors.profileTitle,
   descriptionSelector: selectors.profileDescription,
+  avatarSelector: ".profile__image",
 });
 
+console.log("UserInfo initialized");
+
 // Initialize Section for cards
-const cardListSection = new Section(
-  {
-    items: initialCards,
-    renderer: (item) => {
-      const card = createCard(item);
-      cardListSection.addItem(card);
+let cardSection;
+let userId;
+
+function createCard(cardData) {
+  const card = new Card({
+    data: cardData,
+    cardSelector: selectors.cardTemplate,
+    handleCardClick: (data) => {
+      imagePopup.open(data);
     },
-  },
-  selectors.cardSection
-);
+    handleLikeClick: (cardId, isLiked) => {
+      console.log(
+        `Like clicked for card ${cardId}. Currently liked: ${isLiked}`
+      );
+      const likeMethod = isLiked
+        ? api.unlikeCard.bind(api)
+        : api.likeCard.bind(api);
+
+      likeMethod(cardId)
+        .then((newCardData) => {
+          console.log("New card data after like/unlike:", newCardData);
+          card.updateLikes(newCardData);
+        })
+        .catch((err) => console.error("Error updating like:", err));
+    },
+    handleDeleteClick: (cardId) => {
+      console.log("Delete clicked", cardId);
+      deleteCardPopup.open();
+      deleteCardPopup.setAction(() => {
+        deleteCardPopup.renderLoading(true);
+        api
+          .deleteCard(cardId)
+          .then(() => {
+            card.deleteCard();
+            deleteCardPopup.close();
+          })
+          .catch((err) => console.error("Error deleting card:", err))
+          .finally(() => {
+            deleteCardPopup.renderLoading(false);
+            deleteCardPopup.resetButtonText();
+          });
+      });
+    },
+    userId: userId,
+  });
+  return card.createCard();
+}
+
+function handleProfileFormSubmit(formData) {
+  console.log("Profile form submitted", formData);
+  editProfilePopup.renderLoading(true);
+  const formValidator = formValidators["profile-edit-modal"];
+  if (formValidator) {
+    formValidator.toggleButtonState();
```

Comment:
`.toggleButtonState()` doesn't have any effect before the request and in `finally`. Please, delete them from the whole code of `index.js`.  It doesn't disable the button - it toggles it. But before the request you have valid data in the inputs, so you can't disable the button like that